### PR TITLE
Develop dashboard layout, update API for PAX and GMV

### PR DESCRIPTION
**Pull request by dipto-itransition**
**_Hello team!_**

### Overview
This pull request introduces a redesigned dashboard layout that enhances the user experience by presenting key metrics and actions in a clear and organized manner. The changes include two primary cards for "PAX booked" and "GMV," as well as three additional cards for "Bookings," "Trading," and "Settings."

### Changes Made

**In Dashboard Layout::**

- Added two cards in the first row:
-> PAX Booked: Displays the number of sold tickets with an accompanying delta value.
-> GMV: Shows the total earned amount with a delta value for tracking changes.

- Additional Cards in the second row:
-> Bookings: Includes a description and a button linking to the Bookings page.
-> Trading: Provides a brief overview of trading statistics with a disabled button labeled "Soon."
-> Settings: Offers customization options with a similarly disabled button.

Note: Each card's delta value is represented using an Ant Design Tag component & utilizes unique icons selected from the Ant Design icon library.

**API Update:**
Modified the _/api/dashboard_ and its mock to support a new JSON format, including-
- User name, PAX value, GMV value, and their respective delta values.
- Introduced currency formatting using the Intl API to adhere to the en-US locale standards, ensuring monetary values are presented correctly (USD).

### Testing
- The implementation has been tested with relevant test cases to verify the accuracy of data rendering and button functionality.

### Impact
This update significantly improves the dashboard's usability and aesthetic appeal, enabling users to quickly access important information and navigate efficiently.

Please feel free to share your thoughts and feedback on this update!

Screenshot of Dashboard

### CODE REVIEW

File: src/App.test.tsx [hunk 1]
```diff
@@ -22,7 +23,25 @@ jest.mock('antd', () => {
...
+
+const TestWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
+  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+);
+
 test('renders Home page on initial render', () => {
-  render(<App />);
-  expect(screen.getByText('Home')).toBeInTheDocument();
+  render(
+    <TestWrapper>
```
Comment (alexeylos) [inline src/App.test.tsx]: You don't need to use QueryClient in tests cause' you mock `useDashboardData` so the code in tests doesn't perform any API calls. So you don't need `TestWrapper` here, `render(<App />)` should be enough to work

File: src/components/reusable/FeatureCard.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  return (
+    <div>
+      <Card>
+        <div className="feature-card">
+          <div className="left-content">{icon}</div>
+          <div className="right-content">
+            <div className="title">{title}</div>
+            <div className="description">{description}</div>
+            {renderButton()}
```
Comment (alexeylos) [inline src/components/reusable/FeatureCard.tsx]: You don't need `renderButton` function. Can be just:
```
{buttonLink ? (
              <Button
                className="btn-style"
                type="primary"
                href={buttonLink}
                icon={<LinkOutlined />}
                block={!!buttonLink}
              >
                Go to the Page
              </Button>
            ) : (
              <Button className="btn-style" disabled block>
                Soon
              </Button>
            )}
```

File: src/components/reusable/FeatureCard.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { FeatureCardProps } from '@/types/reuseableCardProps';
+
+const FeatureCard: React.FC<FeatureCardProps> = ({ icon, title, description, buttonLink }) => {
+  const renderButton = () => {
+    if (buttonLink) {
+      return (
+        <Button
+          className="btn-style"
+          type="primary"
+          href={buttonLink}
```
Comment (alexeylos) [inline src/components/reusable/FeatureCard.tsx]: nit ('nit' means that this comment is not blocking review, it's just a suggestion or note): I'd suggest to use `onClick` with `useNavigate` liks you've done for `SiderMenu` cause' `navigate` uses seamess navigation without page reload while `href` attribute for Button triggers page reload on navigation

File: src/components/reusable/InfoCard.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import { RiseOutlined, FallOutlined } from '@ant-design/icons';
+
+const InfoCard: React.FC<ReusableCardProps> = ({ title, value, delta, currency }) => {
+  return (
+    <div>
+      <Card className="custom-card">
+        <div className="title">{title}</div>
+        <div className="value">
+          {currency == 'USD' ? (
+            <svg
```
Comment (alexeylos) [inline src/components/reusable/InfoCard.tsx]: nit: I'd suggest to store `svg` in `.svg` files out of JSX

**Review by alexeylos — APPROVED** [review]

File: src/components/reusable/FeatureCard.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import { Card, Button } from 'antd';
+import { LinkOutlined } from '@ant-design/icons';
+import { FeatureCardProps } from '@/types/reuseableCardProps';
+import { useNavigate } from 'react-router-dom';
+import './FeatureCard.less';
+
+const FeatureCard: React.FC<FeatureCardProps> = ({ icon, title, description, buttonLink }) => {
+  const navigate = useNavigate();
+
+  const naviagteUrl = (url: string) => {
```
Comment (alexeylos) [inline src/components/reusable/FeatureCard.tsx]: typo in `navigateUrl`

**Review by mainul-itransition — APPROVED** [review]

**Pull request merged**